src/Compiler/Service/IncrementalBuild.fs

```diff
@@ -235,184 +226,199 @@ type TcInfoExtras =
     member x.TcSymbolUses =
         x.tcSymbolUses
 
-[<AutoOpen>]
-module TcInfoHelpers =
-
-    let emptyTcInfoExtras =
-        {
-            tcResolutions = TcResolutions.Empty
-            tcSymbolUses = TcSymbolUses.Empty
-            tcOpenDeclarations = [||]
-            latestImplFile = None
-            itemKeyStore = None
-            semanticClassificationKeyStore = None
-        }
-
-/// Accumulated results of type checking.
-[<NoEquality; NoComparison>]
-type TcInfoState =
-    | PartialState of TcInfo
-    | FullState of TcInfo * TcInfoExtras
-
-    member x.TcInfo =
-        match x with
-        | PartialState tcInfo -> tcInfo
-        | FullState (tcInfo, _) -> tcInfo
-
-    member x.TcInfoExtras =
-        match x with
-        | PartialState _ -> None
-        | FullState (_, tcInfoExtras) -> Some tcInfoExtras
-
-[<NoEquality; NoComparison>]
-type TcInfoNode =
-    | TcInfoNode of partial: GraphNode<TcInfo> * full: GraphNode<TcInfo * TcInfoExtras>
-
-    member this.TryGetFull =
-        match this with
-        | TcInfoNode(_, full) -> full.TryPeekValue
+module private ValueOption =
```

Comment:
Nitpick but I believe the `private` modifier is redundant if this is not listed in the signature file.